zokrates_core/src/compile.rs

```diff
@@ -186,37 +212,42 @@ pub fn compile<T: Field, E: Into<imports::Error>>(
     location: FilePath,
     resolver: Option<&dyn Resolver<E>>,
     config: &CompileConfig,
-) -> Result<CompilationArtifacts<T>, CompileErrors> {
+) -> Result<BinCompilationArtifacts<ir::Prog<T>>, CompileErrors> {
     let arena = Arena::new();
 
-    let (typed_ast, abi) = check_with_arena(source, location.clone(), resolver, config, &arena)?;
-
-    // flatten input program
-    log::debug!("Flatten");
-    let program_flattened = Flattener::flatten(typed_ast, config);
-
-    // constant propagation after call resolution
-    log::debug!("Propagate flat program");
-    let program_flattened = program_flattened.propagate();
-
-    // convert to ir
-    log::debug!("Convert to IR");
-    let ir_prog = ir::Prog::from(program_flattened);
-
-    // optimize
-    log::debug!("Optimise IR");
-    let optimized_ir_prog = ir_prog.optimize();
-
-    // analyse ir (check constraints)
-    log::debug!("Analyse IR");
-    let optimized_ir_prog = optimized_ir_prog
-        .analyse()
-        .map_err(|e| CompileErrorInner::from(e).in_file(location.as_path()))?;
-
-    Ok(CompilationArtifacts {
-        prog: optimized_ir_prog,
-        abi,
-    })
+    let artifacts = check_with_arena(source, location.clone(), resolver, config, &arena)?;
+
+    match artifacts {
+        CompilationArtifacts::Lib => unreachable!(),
```

Comment:
This might be reachable when called in `zokrates-js` because config is passed by the caller:

https://github.com/Zokrates/ZoKrates/pull/988/files#diff-31d8ede13068a8e0e79ec05ff89244788c8903841d377659d712b5d2057e9ea6R109

If my understanding is correct, one could set the mode to lib and call compile which would throw here with no clear message.